src/v/cluster/controller_backend.cc

```diff
@@ -1125,9 +1163,28 @@ ss::future<result<ss::stop_iteration>> controller_backend::reconcile_ntp_step(
           updated_replicas
           && update_it->second.get_state()
                == reconfiguration_state::force_update) {
-            // For force-update new replica starts with the updated
-            // configuration right away (joint consensus is not used)
-            initial_replicas = *updated_replicas;
+            auto original_vnodes = create_vnode_set(
+              *orig_replicas,
+              p_it->second.replicas_revisions,
+              last_cmd_revision);
+            auto updated_vnodes = create_vnode_set(
+              *updated_replicas,
+              p_it->second.replicas_revisions,
+              last_cmd_revision);
+            auto original_nodes_retained = intersect(
+              original_vnodes, updated_vnodes);
+            // Current nodes is a voter only if we do not
```

Comment:
nit: as the logic is a bit complicated here, perhaps we can factor out the voters/learners calculation from `force_replica_set_update` and use it here. I think the conditional then becomes simply `if (learners.empty())` or equivalently `if (_self in voters)` which is easier to comprehend.